source/server.lua

```diff
@@ -27,12 +30,9 @@ lib.callback.register("ND_AppearanceShops:clothingPurchase", function(src, store
     return true
 end)
 
-RegisterNetEvent("ND_AppearanceShops:updateAppearance", function(clothing)
-    local src = source
-    local player = NDCore.getPlayer(src)
-    if not player then return end
-
-    if clothing and type(clothing) == "table" then
-        player.setMetadata("clothing", clothing)
+AddEventHandler("NDCore:CharacterLoaded", function(player)
+    local clothing = player.getMetadata("clothing")
+    if clothing then
+        TriggerClientEvent("ND_AppearanceShops:ApplyAppearance", player.source, clothing)

```

Comment:
Tis is unnecessary, updateAppearance is used to save the clothing data to the players metadata. Also the character selection menus should be using the character loaded event to apply clothing. Such as in [ND_Characters/source/client.lua#L57-L70](https://github.com/ND-Framework/ND_Characters/blob/main/source/client.lua#L57-L70).